README.md

```diff
@@ -22,3 +29,41 @@ and this to your crate root:
 ```rust
 extern crate uuid;
 ```
+
+## Examples
+
+To create a new random (V4) UUID and print it out in hexadecimal form:
+
+```rust
+use uuid::Uuid;
+
+fn main() {
+    let my_uuid = Uuid::new_v4();
+    println!("{}", my_uuid.to_string());
+}
+```
+
+The library supports 5 versions of UUID:
+
+Name     | Version
+---------|----------
+Mac	     | Version 1: MAC address
+Dce	     | Version 2: DCE Security
+Md5	     | Version 3: MD5 hash
+Random   | Version 4: Random
+Sha1     | Version 5: SHA-1 hash
+
+To create a new UUID by specifying the format and printing a hyphenated string:
+
+```rust
+use uuid::{Uuid, UuidVersion};
+
+fn main() {
+    let my_uuid = Uuid::new(UuidVersion { Sha1 }).unwrap();
+    println!("{}", my_uuid.to_hyphenated_string());
```

Comment:
Hm I don't think this will compile (e.g. `UuidVersion::Sha1` to get the variant) or work unfortunately (`None` is returned for anything other than a random uuid). Perhaps a different example could be used? (such as parsing one)